source/extensions/filters/http/basic_auth/config.cc

```diff
@@ -0,0 +1,69 @@
+#include "source/extensions/filters/http/basic_auth/config.h"
+
+#include "source/common/config/datasource.h"
+#include "source/extensions/filters/http/basic_auth/basic_auth_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace BasicAuth {
+
+using envoy::extensions::filters::http::basic_auth::v3::BasicAuth;
+
+namespace {
+
+UserMap readHtpasswd(std::string htpasswd) {
+  UserMap users;
+  std::istringstream htpsswd_ss(htpasswd);
+  std::string line;
+
+  while (std::getline(htpsswd_ss, line)) {
+    size_t colonPos = line.find(':');
+
+    if (colonPos != std::string::npos) {
+      std::string name, hash;
+
+      name = line.substr(0, colonPos);
+      hash = line.substr(colonPos + 1);
```

Comment:
It would be better use string view here to avoid unnecessary string copy.

Reply:
I'm not an experienced C++ programmer, so please correct me if I'm wrong :-)

I think we need to use string instead of string_view here. We need to store the username and password in the user map, and a string_view will point to a null string because the original string is a local variable.

Reply:
Oh, I see, you mean just use a string_view to avoid copying from the line.